graf2d/gpad/v7/inc/ROOT/TMenuItem.hxx

```diff
@@ -0,0 +1,52 @@
+/// \file ROOT/TMenuItem.h
+/// \ingroup Base ROOT7
+/// \author Sergey Linev
+/// \date 2017-06-29
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2017, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_TMenuItem
+#define ROOT7_TMenuItem
+
+#include <string>
+#include <vector>
+
+namespace ROOT {
+namespace Experimental {
+
+/** \class TMenuItem
+  Class contains info for producing menu item on the JS side.
+  */
+
+class TMenuItem {
+protected:
+   std::string  fName;     //  name of the menu item
+   std::string  fTitle;    //  title of menu item
+   int          fChecked;  // -1 not exists, 0 - off, 1 - on
+   std::string  fExec;     // execute when item is activated
+public:
+
+   TMenuItem() : fName(), fTitle(), fChecked(-1), fExec() {}
```

Comment:
Can you replace that default constructor with an in-class member initialization:
`int   fChecked = -1;` (or the equivalent enum constant once you have it) and then declare the default constructor as `TMenuItem() = default;`?

Like that the compiler will generate the default constructor itself (and the class will be trivial).